ql/cashflows/averagebmacoupon.cpp

```diff
@@ -225,10 +225,10 @@ namespace QuantLib {
             refStart = start = schedule_.date(i);
             refEnd   =   end = schedule_.date(i+1);
             paymentDate = calendar.adjust(end, paymentAdjustment_);
-            if (i == 0 && !schedule_.isRegular(i+1))
+            if (i == 0 && schedule_.hasIsRegular() && !schedule_.isRegular(i+1))
```

Comment:
You need to check for the tenor as well.

Reply:
right, will change it